xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/xwiki.js

```diff
@@ -1383,6 +1383,7 @@ if (htmlElement.readAttribute('data-xwiki-reference') != null) {
   XWiki.Document.currentDocumentReference = XWiki.currentDocument.getDocumentReference();
 }
 XWiki.Document.URLTemplate = "$xwiki.getURL('__space__.__page__', '__action__')";
+XWiki.Document.ViewURLTemplate = "$xwiki.getURL('__space__.__page__', 'view')";
```

Comment:
Thanks for this pull request! You could have also used the ``$xwiki.getURL('__space__.__page__')`` signature (without the action). I would add a short comment (besides the commit message) to explain why we need a separate template for the view URL, something like

    // We need a separate template for the view URL because the action name is missing when Short URLs are enabled.